specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json

```diff
@@ -0,0 +1,650 @@
+    },
+    "ContainerServiceVnetSubnetID": {
+      "type": "string",
+      "description": "VNet SubnetID specifies the vnet's subnet identifier."
+    },
+    "OSType": {
+      "type": "string",
+      "default": "Linux",
+      "enum": [
+        "Linux",
+        "Windows"
+      ],
+      "x-ms-enum": {
+        "name": "OSType",
+        "modelAsString": true
+      },
+      "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+    },
+    "OpenShiftContainerServiceVMSize": {
+      "type": "string",
+      "x-ms-enum": {
+        "name": "OpenShiftContainerServiceVMSize",
+        "modelAsString": true
+      },
+      "description": "Size of OpenShift VMs.",
+      "enum": [
+        "Standard_D2s_v3",
+        "Standard_D4s_v3"
+      ]
+    },
+    "OpenShiftAgentPoolProfileRole": {
+      "type": "string",
+      "enum": [
+        "compute",
+        "infra"
+      ],
+      "x-ms-enum": {
+        "name": "OpenShiftAgentPoolProfileRole",
+        "modelAsString": true
+      },
+      "description": "OpenShiftAgentPoolProfileRole represents the role of the AgentPoolProfile."
+    },
+    "OpenShiftManagedClusterMasterPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique name of the master pool profile in the context of the subscription and resource group."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 10,
+          "minimum": 1,
+          "description": "Number of masters (VMs) to host docker containers. The default value is 3.",
+          "default": 3
+        },
+        "vmSize": {
+          "$ref": "#/definitions/OpenShiftContainerServiceVMSize",
+          "description": "Size of agent VMs."
+        },
+        "vnetSubnetID": {
+          "$ref": "#/definitions/ContainerServiceVnetSubnetID",
+          "description": "VNet SubnetID specifies the vnet's subnet identifier."
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        }
+      },
+      "required": [
+        "count",
+        "vmSize"
+      ],
+      "description": "OpenShiftManagedClusterMaterPoolProfile contains configuration for OpenShift master VMs."
+    },
+    "OpenShiftManagedClusterAgentPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique name of the pool profile in the context of the subscription and resource group."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 5,
+          "minimum": 1,
+          "description": "Number of agents (VMs) to host docker containers. Allowed values must be in the range of 1 to 5 (inclusive). The default value is 1. ",
+          "default": 1
```

Comment:
I quick comment on the use of "defaults". I see you have a couple of them in the spec, so I wont leave a comment for every one.
It's important to note that how "defaults" are handled can vary from language to language.
Some languages will send the default values in the request body (regardless of whether it's set by the user) and other languages will leave these fields in the request as empty, and use the default purely to document the service behavior.

So please make sure that your service API will handle both scenarios the same way - i.e. it will treat a request with an empty field and a request with a default value exactly the same.